### PR TITLE
Edit pwd and name

### PR DESCRIPTION
**Pull request by J4ken**

### CODE REVIEW

File: app/controllers/user_controller.rb [hunk 1]
```diff
@@ -74,6 +77,42 @@ def reset_password
...
+    if request.post?
+      response = database.edit_password(
+          params[:current_password],
+          params[:new_password],
+          params[:password_confirmation]
+      )
+
+      if response.success?
+        flash[:success] = 'Ditt lösenord har uppdaterads'
+        redirect_to root_url
```
Comment (patriksletmo) [inline app/controllers/user_controller.rb]: Should redirect to edit_password_url.

File: app/controllers/user_controller.rb [hunk 2]
```diff
@@ -74,6 +77,42 @@ def reset_password
...
+      else
+        flash[:error] = response.friendly_error
```
Comment (patriksletmo) [inline app/controllers/user_controller.rb]: Should use flash.now since no redirection is made.

File: app/controllers/user_controller.rb [hunk 3]
```diff
@@ -74,6 +77,42 @@ def reset_password
...
+          current_user['id'],
+          {user:{display_name: params[:display_name]
+          }}
+      )
+
+      if response.success?
+        flash[:success] = 'Ditt namn har updaterats'
+        redirect_to edit_name_url
+      else
+        flash[:error] = response.friendly_error
```
Comment (patriksletmo) [inline app/controllers/user_controller.rb]: Use flash.now.

File: app/controllers/user_controller.rb [hunk 4]
```diff
@@ -74,6 +77,42 @@ def reset_password
...
+      end
+    end
+  end
+
+  def edit_name
+    if request.post?
+      response = database.update_user(
+          current_user['id'],
+          {user:{display_name: params[:display_name]
```
Comment (patriksletmo) [inline app/controllers/user_controller.rb]: Place all braces on the same line or format this differently.

File: app/views/user/edit_name.html.erb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+          <%= hidden_field_tag :authenticity_token, form_authenticity_token %>
+          <div class="card-content">
+            <span class="card-title">Byta namn</span>
+            <div class="input-field">
+              <input type="text" name="display_name" id="display_name" required>
+              <label for="display_name">Nytt namn</label>
+            </div>
+          </div>
+            <div class="card-action">
+              <a class="blue-text" href="javascript:;" onclick="document.getElementById('edit-display_name-form').submit();">BYT!</a>
```
Comment (patriksletmo) [inline app/views/user/edit_name.html.erb]: Should .click() instead of .submit()

File: app/views/user/edit_password.html.erb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+              <input type="password" name="new_password" id="new_password" required>
+              <label for="password">Nytt Lösenord</label>
+            </div>
+            <div class="input-field">
+              <input type="password" name="password_confirmation" id="password_confirmation" required>
+              <label for="password_confirmation">Upprepa lösenord</label>
+            </div>
+
+            <div class="card-action">
+              <a class="blue-text" href="javascript:;" onclick="document.getElementById('edit-password-form').submit();">Skicka</a>
```
Comment (patriksletmo) [inline app/views/user/edit_password.html.erb]: Same as above regarding .click()

File: config/routes.rb [hunk 7]
```diff
@@ -6,7 +6,9 @@
   get '/lang/:lang', to: 'application#change_language'
 
   # Authentication
-  get '/profile', to: 'user#index'
+  match '/profile', to: 'user#index', via: [:get, :post]
```
Comment (patriksletmo) [inline config/routes.rb]: Should only allow get requests.

File: app/controllers/user_controller.rb [hunk 8]
```diff
@@ -1,6 +1,9 @@
 class UserController < NavigationController
   def index
     return if require_login!
+
```
Comment (ljukas) [inline app/controllers/user_controller.rb]: Remove empty lines.

File: app/database/areas/account_area.rb [hunk 9]
```diff
@@ -35,6 +35,15 @@ def reset_password(new_password, confirmation)
       put('/api/v1/auth/password', options.merge(authenticated_options))
     end
 
+    def edit_password(current_password, new_password, confirmation)
+      options = {:body => {
```
Comment (ljukas) [inline app/database/areas/account_area.rb]: Formatting, put :body on a new line

File: app/views/user/index.html.erb [hunk 10]
```diff
@@ -2,4 +2,15 @@
...
 <div class="container">
   <p>Inloggad som <%= current_user['display_name'] %></p>
+
+  <% if current_user['provider'] != 'cas' %>
+    <a href="<%= url_for action: :edit_password%>" class="btn">
+      Byt lösenord
+    </a>
+  <% end %>
+
+  <a href="<%= url_for action: :edit_name%>" class="btn">
```
Comment (ljukas) [inline app/views/user/index.html.erb]: Använd hellre en link_to instället för att kombinera erb med html

File: app/database/areas/account_area.rb (hunk 9 above)
Comment (J4ken) [inline app/database/areas/account_area.rb]: Just following the previous formatting pattern

**Pull request merged**